my_each.rb

```diff
@@ -1,6 +1,141 @@
-module Enumerable
-    def my_each(array)
-      for i in 0...array.length
-        p "gaa"
+module Enumerable # rubocop:disable Style/ModuleLength
+  def my_each
+    array = to_a
+    i = 0
+    while i < array.size
+      return to_enum(:my_each) unless block_given?
+
+      yield array[i]
+      i += 1
+
     end
+
+    self
+  end
+
+  def my_each_with_index
+    array = to_a
+    i = 0
+    while i < array.size
+      return to_enum(:my_each_with_index) unless block_given?
+
+      yield array[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    new_array = []
+    my_each do |element|
+      return to_enum(:my_select) unless block_given?
+
+      new_array.push(element) if yield element
+    end
+    new_array
+  end
+
+  def my_all?(compare = nil)
+    result = true
+    my_each do |element|
+      result = if compare
+                 compare.class.eql?(Class) ? element.is_a?(compare) : element.to_s.match?(compare)
+               elsif block_given?
+                 yield element
+
+               else
+                 element ^ 1
+               end
+      break unless result
+    end
+
+    result
+  end
+
+  def my_any?(compare = nil)
+    result = false
+    my_each do |element|
+      result = if compare
+                 compare.class.eql?(Class) ? element.is_a?(compare) : element.to_s.match?(compare.to_s)
+               elsif block_given?
+                 yield element
+               else
+                 element ^ 1
+               end
+      break if result
+    end
+    result
+  end
```

Comment:
* [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
  ```ruby
    array = [nil, false, true, []]
    p array.my_any? == array.any? # false
  ```

Reply:
I changed my_all and my_none, but this method was already working